packages/pluginutils/src/dataToEsm.ts

```diff
@@ -91,7 +91,7 @@ const dataToEsm: DataToEsm = function dataToEsm(data, options = {}) {
 
   let maxUnderbarPrefixLength = 0;
   for (const key of Object.keys(data)) {
-    const underbarPrefixLength = key.match(/^(_+)/)?.[0].length ?? 0;
+    const underbarPrefixLength = /^(_+)/.exec(key)?.[0].length ?? 0;
```

Comment:
I would expect here and in the other that perhaps it'd be better to move the regex declaration outside the method declaration so that it doesn't have to be recompiled each method invocation and each iteration of this loop

Reply:
Something I've found over time is that it doesn't necessarily help with perf by hoisting it out. Engines these days are able to optimize it, and in most cases only yield very small gains. This was also discussed at https://github.com/lukeed/polka/pull/210

Reply:
the discussion there suggests the one place it matters is in loops, so it might still be worth changing here, but not in the other locations?

Reply:
Yeah perhaps we can try that here, but I'd prefer to have a benchmark on that first before making the change though